tests/unit/test_io_hdf5_h5tools.py

```diff
@@ -1017,39 +1071,35 @@ def test_io_read_herd_value_warn(self):
 
         self.remove_er_files()
 
-    def test_io_write_herd(self):
-        er = HERD()
-        self.foofile.link_resources(er)
-
-        data = Data(name="species", data=['Homo sapiens', 'Mus musculus'])
-        er.add_ref(file=self.foofile,
-                   container=data,
-                   key='key1',
-                   entity_id='entity_id1',
-                   entity_uri='entity1')
+    @unittest.skipIf(not LINKML_INSTALLED, "optional LinkML module is not installed")
+    def test_io_write_extend_herd(self):
+        """
+        Test the optional write of HERD with extending an existing HERD instance.
+        """
+        terms = TermSet(term_schema_path='tests/unit/example_test_term_set.yaml')
+        foo = Foo(name="species", attr1='attr1', attr2=0,
+                  my_data=TermSetWrapper(value=['Homo sapiens', 'Mus musculus'],
+                                                         termset=terms))
 
-        with HDF5IO(self.path, manager=self.manager, mode='w', herd_path='./HERD.zip') as io:
-            io.write(self.foofile)
+        foobucket = FooBucket('bucket1', [foo])
+        foofile = FooFile(buckets=[foobucket])
 
-        with HDF5IO(self.path, manager=self.manager, mode='r', herd_path='./HERD.zip') as io:
-            container = io.read()
-            self.assertIsInstance(io.herd, HERD)
-            self.assertIsInstance(container.get_linked_resources(), HERD)
+        er = HERD()
+        er.add_ref(file=foofile,
+                   container=foofile,
+                   key='special',
+                   entity_id="id11",
+                   entity_uri='url11')
 
-        self.remove_er_files()
+        with HDF5IO(self.path, manager=get_foo_buildmanager("text"), mode='w', herd_path='./HERD.zip') as io:
+            io.write(foofile)
 
-    def test_io_warn(self):
-        er = HERD()
+        with HDF5IO(self.path, manager=get_foo_buildmanager("text"), mode='r') as io:
+            read_foofile = io.read()
+            self.assertListEqual(foofile.buckets['bucket1'].foos['species'].my_data.value,
+                                 read_foofile.buckets['bucket1'].foos['species'].my_data[:].tolist())
 
-        data = Data(name="species", data=['Homo sapiens', 'Mus musculus'])
-        er.add_ref(file=self.foofile,
-                   container=data,
-                   key='key1',
-                   entity_id='entity_id1',
-                   entity_uri='entity1')
-        with HDF5IO(self.path, manager=self.manager, mode='w', herd_path='./HERD.zip') as io:
-            with self.assertWarns(Warning):
-                io.write(self.foofile)
+        self.remove_er_files()
```

Comment:
Can you test/assert whether the ref gets written in the HERD file

Can you also test whether passing HERD to `io.write` works?

Reply:
The extend is in the tests for write. I did add more asserts now to check the HERD itself.

Reply:
> The extend is in the tests for write.

Can you point me to the test?